### PR TITLE
feat(anvil): enable IPC for Windows

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

blocked by ~~gakonst/ethers-rs#1976~~, ~~#3977~~

### CODE REVIEW

File: anvil/Cargo.toml [hunk 1]
```diff
@@ -83,3 +83,6 @@ crc = "3.0.0"
 default = ["cli"]
 cmd = ["foundry-common", "forge", "clap", "clap_complete", "ctrlc", "anvil-server/clap"]
 cli = ["tokio/full", "cmd", "fdlimit"]
+
+[patch."https://github.com/gakonst/ethers-rs"]
+ethers = { git = "https://github.com/danipopes/ethers-rs", branch = "feat/named-pipes-provider" }
```
Comment (mattsse) [inline anvil/Cargo.toml:88]: please patch at root Cargo.toml, patches in member crates are ignored

**Review by mattsse — CHANGES_REQUESTED** [review]

**Comment by DaniPopes** [comment]
image

**Comment by mattsse** [comment]
yeh you can switch to push and it should run

https://github.com/foundry-rs/foundry/blob/848bb445481bf9a29f9aeb47ff4488c31c7aa7bd/.github/workflows/cross-platform.yml#L2-L3

**Comment by DaniPopes** [comment]
opened #3977

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -212,6 +212,5 @@ jobs:
 
     cross-platform:
         name: Cross-platform tests
-        if: github.event_name != 'pull_request'
```
Comment (gakonst) [inline .github/workflows/test.yml:215]: why did we use this?

File: Cargo.toml [hunk 3]
```diff
@@ -56,7 +56,16 @@ panic = "abort"
 codegen-units = 1
 
 # # Patch ethers-rs with a local checkout then run `cargo update -p ethers`
-# [patch."https://github.com/gakonst/ethers-rs"]
+[patch."https://github.com/gakonst/ethers-rs"]
```
Comment (gakonst) [inline Cargo.toml:59]: PR is merged now

File: anvil/tests/it/main.rs [hunk 4]
```diff
@@ -7,7 +7,6 @@ mod ganache;
 mod gas;
 mod genesis;
 mod geth;
-#[cfg(not(windows))]
```
Comment (gakonst) [inline anvil/tests/it/main.rs:10]: nice :)

File: anvil/src/lib.rs [hunk 5]
```diff
@@ -270,10 +270,9 @@ impl NodeHandle {
     }
 
     /// Connects to the ipc endpoint of the node, if spawned
-    #[cfg(not(windows))]
```
Comment (gakonst) [inline anvil/src/lib.rs:273]: 🥳

**Review by gakonst — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (DaniPopes) [reply .github/workflows/test.yml:215]: Maybe to not get spammed? I removed it just so we can test the CI here in this pr

**Comment by DaniPopes** [comment]
**also bumps [`gakonst/ethers-rs` to `4fd742f8cea5fd4de1577f1a8541f8f9c4240ceb`](https://github.com/gakonst/ethers-rs/commit/4fd742f8cea5fd4de1577f1a8541f8f9c4240ceb)

**Comment by gakonst** [comment]
lint failing

**Review by mattsse — APPROVED** [review]

**Pull request merged**